src/main/java/org/sqlite/util/OSInfo.java

```diff
@@ -190,7 +192,7 @@ static String resolveArmArchType() {
             } else if (armType.startsWith("armv5")) {
                 // Use armv5, soft-float ABI
                 return "arm";
-            } else if (armType.startsWith("aarch64")) {
+            } else if (armType.startsWith("aarch64") && !is32bit) {
```

Comment:
by skipping this check the code will try the other methods, but it's not guaranteed that the correct type would be returned.

the following code could return `armv7` or `arm`, but not `armv6` for instance. As we have different `.so` files for each, it may not always work, as it could return an incorrect architecture.

It seems a better way to fix this would be to use something different than `uname -m` to get the arch.

Reply:
I have updated the code to always return `armv7` when running on a 32-bit JVM on an `aarch64` architecture.

I think that all `aarch64` architectures (if they support 32-bit instructions) will always support `armv7`.

Reply:
> I think that all aarch64 architectures (if they support 32-bit instructions) will always support armv7.

i think that's acceptable indeed.